lib/ain-evm/src/core.rs

```diff
@@ -224,11 +225,9 @@ impl EVMCoreService {
                 return Err(anyhow!("insufficient balance to pay fees").into());
             }
 
-            if gas_limit < MIN_GAS_PER_TX {
-                debug!("[validate_raw_tx] gas limit is below the minimum gas per tx");
-                return Err(anyhow!("gas limit is below the minimum gas per tx").into());
-            }
-        } else if balance < MIN_GAS_PER_TX || balance < prepay_fee {
+            // Validate tx gas limit with intrinsic gas
+            check_tx_intrinsic_gas(&signed_tx)?;
+        } else if balance < MIN_GAS_PER_TX.into() || balance < prepay_fee {
```

Comment:
I think this check is not required since it is a subset of the intrinsic gas check.

The call cost is added [here](https://docs.rs/evm-gasometer/0.39.0/src/evm_gasometer/lib.rs.html#230) and the gas check is done [here](https://docs.rs/evm-gasometer/0.39.0/src/evm_gasometer/lib.rs.html#260).

Reply:
Thats incorrect. I think you are confusing gas and fees. This validation checks that the account sender has enough dfi in the account to pay for the prepay gas fees.

Reply:
The unit for `MIN_GAS_PER_TX` is gas and the unit for `balance` is wei from what I see. I don't think this comparison makes sense, IMO only the second check should be in there.

Reply:
that is erroneous pipeline forguarded with changiintermediate, which will be removed in beta6.

Reply:
`
// Changi intermediate constant

pub const MIN_GAS_PER_TX: u64 = 21_000;`

Reference gas.rs::line 12-13

Reply:
Ah makes sense in that case :disappointed:

Reply:
Resolving.